Fails to build "lxml-version.h: No such file or directory"

Suddenly failed to build recently. Might be due to the new release of lxml that came out

```#11 22.22   In file included from /tmp/pip-install-w06wmd73/xmlsec_b455cf60a688463dbc08380657687275/src/ds.c:15:
#11 22.22   /tmp/pip-install-w06wmd73/xmlsec_b455cf60a688463dbc08380657687275/src/lxml.h:18:10: fatal error: lxml-version.h: No such file or directory
#11 22.22      18 | #include <lxml-version.h>
#11 22.22         |          ^~~~~~~~~~~~~~~~
#11 22.22   compilation terminated.
#11 22.22   error: command 'gcc' failed with exit status 1
```

--- a/.github/workflows/linuxbrew.yml
+++ b/.github/workflows/linuxbrew.yml
@@ -18,11 +18,11 @@ jobs:
           echo "eval \$($(brew --prefix)/bin/brew shellenv)" >>~/.profile
           brew update
           brew install python gcc libxml2 libxmlsec1 pkg-config
-          pip3 install --upgrade setuptools wheel
+          pip3 install --upgrade setuptools wheel build
           ln -s $(brew --prefix)/bin/gcc-11 $(brew --prefix)/bin/gcc-5
       - name: Build linux_x86_64 wheel
         run: |
-          python3 setup.py bdist_wheel
+          python3 -m build
           rm -rf build/
       - name: Install test dependencies
         run: |

--- a/.github/workflows/macosx.yml
+++ b/.github/workflows/macosx.yml
@@ -14,7 +14,7 @@ jobs:
           python-version: ${{ matrix.python }}
       - name: Install build dependencies
         run: |
-          pip install --upgrade pip setuptools wheel
+          pip install --upgrade pip setuptools wheel build
           brew install libxml2 libxmlsec1 pkg-config
       - name: Build macosx_x86_64 wheel
         env:
@@ -22,7 +22,7 @@ jobs:
           CFLAGS: "-fprofile-instr-generate -fcoverage-mapping"
           LDFLAGS: "-fprofile-instr-generate -fcoverage-mapping"
         run: |
-          python setup.py bdist_wheel
+          python -m build
           rm -rf build/
       - name: Set environment variables
         shell: bash

--- a/.github/workflows/manylinux2010.yml
+++ b/.github/workflows/manylinux2010.yml
@@ -11,7 +11,7 @@ jobs:
       - uses: actions/checkout@v1
       - name: Install build dependencies
         run: |
-          /opt/python/${{ matrix.python-abi }}/bin/pip install --upgrade pip setuptools wheel
+          /opt/python/${{ matrix.python-abi }}/bin/pip install --upgrade pip setuptools wheel build
       - name: Set environment variables
         shell: bash
         run: |
@@ -22,7 +22,7 @@ jobs:
           # disable libxml2-2.9.12 because of https://gitlab.gnome.org/GNOME/libxslt/-/issues/52
           PYXMLSEC_LIBXML2_VERSION: 2.9.10
         run: |
-          /opt/python/${{ matrix.python-abi }}/bin/python setup.py bdist_wheel
+          /opt/python/${{ matrix.python-abi }}/bin/python -m build
       - name: Label manylinux2010_x86_64 wheel
         run: |
           ls -la dist/

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,4 +23,4 @@ known_first_party = ['xmlsec']
 known_third_party = ['lxml', 'pytest', '_pytest', 'hypothesis']
 
 [build-system]
-requires = ['setuptools>=42', 'wheel', 'setuptools_scm[toml]>=3.4']
+requires = ['setuptools>=42', 'wheel', 'setuptools_scm[toml]>=3.4', "pkgconfig>=1.5.1", "lxml>=3.8, !=4.7.0"]

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,2 @@
 pkgconfig
-lxml >= 3.8.0
+lxml >= 3.8.0, !=4.7.0

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,2 +1,1 @@
-pkgconfig
 lxml >= 3.8.0, !=4.7.0

